master/security/service-accounts.md

```diff
@@ -0,0 +1,95 @@
+
+Using **Istio-enabled apps** with Calico network policy, the cryptographic identity associated with the service account is checked (along with the network identity) to achieve two-factor authentication.
+
+### Features
+
+This how-to guide uses the following Calico features:
+
+**NetworkPolicy** or **GlobalNetworkPolicy** with a service account rule and match criteria.
+
+
+### Concepts
+
+#### Use smallest set of permissions required
+
+Service accounts are controlled by RBAC so you can grant permissions to trusted entities (code and/or people). To perform any operation in a workload, clients are required to authenticate with the Kubernetes API server. If you do not explicitly assigned a service account to a pod, it uses the default ServiceAccount in the namespace. You should not grant broad permissions to the default service account for a namespace. If an application needs access to the Kubernetes API, create separate service accounts with the smallest set of permissions required. 
+
+#### Service account labels
+
+Like all other Kubernetes objects, service accounts have labels. So you can use labels to create ‘groups’ of service accounts. For Calico network policy, you can limit access to workloads by matching:
+
+- Endpoints in a service account (additionally, with a positive match on a namespace)
+- Service account names and selectors
+
+### Before you begin...
+
+Configure unique Kubernetes service accounts for your applications
+
+### How to
+
+- [Limit ingress traffic for workloads by service account name](#limit-ingress-traffic-for-workloads-by-service-account-name)
+- [Limit ingress traffic for workloads by service account label](#limit-ingress-traffic-for-workloads-by-service-account-label)
+
+#### Limit ingress traffic for workloads by service account name
+
+In the following example, ingress traffic is allowed from any workload in the **prod-engineering** namespace, whose service account has the names **ingress-sa-in-calico** or **ingress-w-in-calico**.
+
+```
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata: 
+  name: demo-calico
+  namespace: prod-engineering
+spec: 
+  ingress: 
+    - action: Allow
+      source: 
+        namespaceselector: 'ingress == "prod-engineering"'
+        selector: 'app == "db"'
+        serviceAccounts: 
+          names: 
+            - ingress-sa-in-calico
+            - ingress-w-in-calico
+  selector: 'app == "front-end"'
+```
+
+#### Limit ingress traffic for workloads by service account label
+
+In the following example, ingress traffic is allowed from any workload in the **prod-engineering** namespace, whose service account has the label **app == db**.
+
+```
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata: 
+  name: demo-calico
+  namespace: prod-engineering
+spec: 
+  ingress: 
+    - action: Allow
+      source: 
+        serviceAccounts: 
+          names: 
```

Comment:
We should remove `names:` from this example

Reply:
Fixed.